### ISSUE TITLE
osmc text labels not reasonably possible (unsupported)

### ISSUE THREAD

**Issue by gorn** [lonvia/waymarked-trails-site#242]
Currently there is no way to specify symbols consisting only of text (or single character). For example
tag 

`red:white:R:red`

should render like red letter R on white background. But it renders as a word red. 

### Problem analysis
I think the problem is (as documented on http://hiking.waymarkedtrails.org/help/rendering/osmc) that you require the format 

`osmc:symbol=waycolor:background:foreground:text:textcolor`

but the in specification (http://wiki.openstreetmap.org/wiki/Key:osmc:symbol#Label_text_color) states that the foreground part is optional. Although I personally understand that it is easier to have the argument obligatory, it is kind of broken as there is no way to specify label which does NOT have any foreground except the text. And even if one wanted to specify it in the way your rendering supports it, there is no way to do so, because there is no "none" symbol to write things like that red:white:none:R:red 

**Example:** http://www.openstreetmap.org/relation/5503238

**Comment by lonvia** [comment]
Just leave foreground blank and it will render as expected. For your example above, you'd need to use `red:white::R:red`. Example: http://hiking.waymarkedtrails.org/#route?id=2314577

**Comment by gorn** [comment]
Ok, good workaround. However this is not what is on most of the places. If you look at http://wiki.openstreetmap.org/wiki/Key:osmc:symbol specification, the ":"  symbol is clearly part of the optional group containing foreground.

**Comment by Hufkratzer** [comment]
Above mentioned example http://www.openstreetmap.org/relation/5503238 is a **bicycle** route and I thought waymarkedtraisl currently does not support osmc:symbol for them (see closed issue https://github.com/lonvia/waymarked-trails-site/issues/223), strage that it labels it with "red" (http://cycling.waymarkedtrails.org/#route?id=5503238)

**Comment by lonvia** [comment]
I hadn't noticed that we are talking about cycling routes. No, they don't do the osmc:symbol rendering but they do use the text part (i.e.the 4th part) of the osmc:symbol tag if there is no ref tag available.

**Comment by gorn** [comment]
Aha, so maybe this is a wrong assumption - you can not count on the fact that the label is in the 4th part ...

**Comment by gorn** [comment]
I fact if you take the syntax strictly, than the label is the penultimate part whenewer there are more then three parts. If there are three there is no lablel.

**Comment by Hufkratzer** [comment]
I agree to @gorn. I don't think it is possible to change the allowed values anymore. The description on the wiki page may be corrected. Anyway, the rule that simply counts the parts seems to be a good idea, and iIf it works waymarkedtrails should interpret osmc:symbols without the discussed colon correctly (in the sense of the wiki page). Not every osmc:symbol producer knows that waymarkedtrails wants the additional colon.

**Comment by lonvia** [comment]
I've removed the code that looks at the osmc:symbol tag during simple label generation. It was a relic from times when osmc:symbol tags without foreground were not supported at all. Things should sort itself out gradually when relations are changed.

The wiki page used to have the syntax still used by waymarkedtrails but was changed later without discussion to the confusing schema it contains now and I have taken the liberty not to follow the maker of the Reit- and Wanderkarte there. The syntax understood by waymarkedtrails is still a valid subset of the current definition on the wiki page and covers about 98% of the relations. At the moment, that's good enough.
